shared/src/main/java/com/couchbase/lite/MutableDocument.java

```diff
@@ -51,7 +52,7 @@ public MutableDocument() {
      * @param id the document ID.
      */
     public MutableDocument(String id) {
-        super(null, id != null ? id : createUUID(), null);
+        super(null, id != null ? id : createUUID(), (C4Document)null);
```

Comment:
In Java, do we have something like this? 
```suggestion
        super(null, id ?: createUUID(), (C4Document)null); 
        // OR 
        super(null, id ?? createUUID(), (C4Document)null); 
```